Sensor/toon.py

```diff
@@ -7,19 +7,29 @@
 
 from homeassistant.helpers.entity import Entity
 import custom_components.toon as toon_main
+import datetime as datetime
 
 _LOGGER = logging.getLogger(__name__)
 
+STATE_ATTR_DEVICE_TYPE = "device_type"
+STATE_ATTR_LAST_CONNECTED_CHANGE = "last_connected_change"
+
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup sensors."""
+    _toon_main = hass.data[toon_main.TOON_HANDLE]
+
     add_devices([
-        ToonSensor(hass, 'Gas_current', 'CM3'),
-        ToonSensor(hass, 'Gas_today', 'M3'),
         ToonSensor(hass, 'Power_current', 'Watt'),
         ToonSensor(hass, 'Power_today', 'kWh')
     ])
-    _toon_main = hass.data[toon_main.TOON_HANDLE]
+
+    if _toon_main.gas:
```

Comment:
That is a different thing. On the main module you reference the toon config which is the object you get from HA about the configuration.yaml. There you set a gas variable. 
Here you query the actual toon object for the gas attribute which is a named tuple so this will always be true.